### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by fernandorpm**
# React & Redux group project - Space Travelers' Hub
> In this project we had to build a React-Redux web app that allowed the user to reserve rockets and join missions. The user can also check his reservations and missions on the `Profile` page.

## Video ⏺️ 
- https://www.loom.com/share/be118b8dea0d4b1ea87545027378dd28

## To add ✅ 
- React, Redux, Jest Structure
- `missions` directory containing all files related to the missions
- `rockets` directory containing all files related to the missions
- `pages` directory containing all files related to the pages
- `shared` directory containing all files shared by the entire application

## To update 🔢 
- README.md

## To replace ⚠️ 
- `app_screenshot.png` to `app_screenshot.gif`

## Requirements 📓 
- Config & basic setup of React & Redux
- Fetch data and update Redux store
- Render UI: lists
- Write actions and reducers for booking rockets/dragons and joining missions
- Write actions and reducers for canceling rockets/dragons and leaving missions
- Render UI: conditional components rendering
- Render UI: My Profile section
- Document the project
- Make a video

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/__snapshots__/Profile.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`renders correctly 1`] = `
+<div
+  className="mainContainer"
+>
+  <section
+    className="missionsContainer"
```
Comment (mikenath223) [inline src/tests/__snapshots__/Profile.test.js.snap:8]: - [x] This snapshot test fails, kindly fix it.

File: src/tests/__snapshots__/MissionContainer.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`renders correctly 1`] = `
+<table
+  className="mainContainer"
+  style={Object {}}
+>
+  <tr
+    className="headers"
```
Comment (mikenath223) [inline src/tests/__snapshots__/MissionContainer.test.js.snap:9]: - [x] This snapshot test also fails, please fix it.

File: src/tests/RocketContainer.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+import RocketContainer from '../components/rockets/rocketContainer/RocketContainer';
+import store from '../redux/configureStore';
+
+it('renders correctly', () => {
+  const tree = renderer
+    .create(<Provider store={store}><RocketContainer /></Provider>)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (mikenath223) [inline src/tests/RocketContainer.test.js:11]: - [ ] You have written only snapshot tests kindly write unit tests also to increase the test coverage.

File: src/pages/Profile.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import styles from './profileStyle.module.scss';
+
+const container = {
+  initial: {},
+  animate: {
+    transition: {
+      staggerChildren: 0.1,
+      delayChildren: 0.3,
+    },
+  },
```
Comment (mikenath223) [inline src/pages/Profile.js:13]: - [x] You've done pretty well making cool animations and effects.

File: src/redux/missions/missionsSlice.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchAllMissions = createAsyncThunk('missions/fetchAllMissions', async () => {
+  const response = await axios.get('https://api.spacexdata.com/v3/missions');
+  const newArr = response.data.map((miss) => ({
+    mission_name: miss.mission_name,
+    mission_id: miss.mission_id,
```
Comment (mikenath223) [inline src/redux/missions/missionsSlice.js:9]: - [x] Good job also using the redux-toolkit.

**Comment by fernandorpm** [comment]
# To the next reviewer

## About UNIT TESTING
In this project, it wasn't necessary to use any method created by us. The closest that we did to creating methods was to use the Javascript inbuilt method `filter`, which doesn't make sense to test.

And all of our functions were built inside of functional components - which according to how javascript itself works, they can't be called individually.

Even when we try to use test libraries such as `Enzyme`, you can notice that it is not possible, as pointed out in [this Github issue](https://github.com/enzymejs/enzyme/issues/2287).

It is also worth mentioning that there weren't any cards related to testing, or to point out what features we could/should have done to test them later on!

Let us know if you have read this by leaving a thumbs up! 👍 
Thanks in advance!

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**